cpp/dolfinx/common/IndexMap.cpp

```diff
@@ -377,6 +369,12 @@ compute_submap_indices(const IndexMap& imap,
     submap_ghost = std::move(submap_ghost1);
   }
 
+  // Compute submap destination ranks
+  // FIXME Remove call to NBX
```

Comment:
How much work to fix this now? We need to guard against unnecessary NBX calls.

Reply:
It will need a bit of thought to do it in a tidy and efficient way and to check for corner cases. I'm probably not going to have time to look at it until next week. I think it's probably best to merge this fix in the meantime and make an issue to remove NBX, but it's up to you

Reply:
Could you create an issue so it doesn't get forgotten?

Reply:
Done, see https://github.com/FEniCS/dolfinx/issues/3100

Reply:
> How much work to fix this now? We need to guard against unnecessary NBX calls.

As far as I've seen, the NBX calls take very little time compared to almost any other function we have in DOLFINx, so I wouldn't be too worried. But in general I agree that if we can avoid it by using existing comms we should.

Reply:
> > How much work to fix this now? We need to guard against unnecessary NBX calls.
> 
> As far as I've seen, the NBX calls take very little time compared to almost any other function we have in DOLFINx, so I wouldn't be too worried. But in general I agree that if we can avoid it by using existing comms we should.

It is a performance concern at scale with a large number of MPI ranks. NBX should be used only when there is no reasonable alternative.